code/modules/mob/living/simple_animal/hostile/terror_spiders/terror_spiders.dm

```diff
@@ -0,0 +1,327 @@
+		var/obj/machinery/door/firedoor/F = target
+		if(F.density)
+			if(F.blocked)
+				to_chat(src, "The fire door is welded shut.")
+			else
+				visible_message("<span class='danger'>\The [src] pries open the firedoor!</span>")
+				F.open()
+		else
+			to_chat(src, "Closing fire doors does not help.")
+	else if(istype(target, /obj/machinery/door/airlock))
+		var/obj/machinery/door/airlock/A = target
+		if(A.density)
+			try_open_airlock(A)
+	else if(isliving(target))
+		var/mob/living/G = target
+		if(issilicon(G))
+			G.attack_animal(src)
+			return
+		else if(G.reagents && (iscarbon(G)))
+			var/can_poison = 1
+			if(istype(G, /mob/living/carbon/human/))
+				var/mob/living/carbon/human/H = G
+				if(!(H.species.reagent_tag & PROCESS_ORG) || (H.species.flags & NO_POISON))
+					can_poison = 0
+			spider_specialattack(G,can_poison)
+		else
+			G.attack_animal(src)
+	else
+		target.attack_animal(src)
+
+/mob/living/simple_animal/hostile/poison/terror_spider/proc/spider_specialattack(mob/living/carbon/human/L, var/poisonable)
+	L.attack_animal(src)
+
+
+
+
+
+
+// --------------------------------------------------------------------------------
+// --------------------- TERROR SPIDERS: PROC OVERRIDES ---------------------------
+// --------------------------------------------------------------------------------
+
+
+/mob/living/simple_animal/hostile/poison/terror_spider/examine(mob/user)
+	..()
+	var/list/msgs = list()
+	if(stat == DEAD)
+		msgs += "<span class='notice'>It appears to be dead.</span>\n"
+	else
+		if(key)
+			msgs += "<span class='warning'>Its eyes regard you with a curious intelligence.</span>"
+		if(health > (maxHealth*0.95))
+			msgs += "<span class='notice'>It is in excellent health.</span>"
+		else if(health > (maxHealth*0.75))
+			msgs += "<span class='notice'>It has a few injuries.</span>"
+		else if(health > (maxHealth*0.55))
+			msgs += "<span class='warning'>It has many injuries.</span>"
+		else if(health > (maxHealth*0.25))
+			msgs += "<span class='warning'>It is barely clinging on to life!</span>"
+		else if(health < maxHealth && regen_points > regen_points_per_kill)
+			msgs += "<span class='notice'>It appears to be regenerating quickly.</span>"
+		if(killcount >= 1)
+			msgs += "<span class='warning'>It has blood dribbling from its mouth.</span>"
+	to_chat(usr,msgs.Join("<BR>"))
+
+
+/mob/living/simple_animal/hostile/poison/terror_spider/New()
+	..()
+	ts_spiderlist += src
+	add_language("Spider Hivemind")
+	add_language("Galactic Common")
+	default_language = all_languages["Spider Hivemind"]
+
+	web_action = new()
+	web_action.Grant(src)
+	wrap_action = new()
+	wrap_action.Grant(src)
+
+	name += " ([rand(1, 1000)])"
+	msg_terrorspiders("[src] has grown in [get_area(src)].")
+	if(is_away_level(z))
+		spider_awaymission = 1
+		ts_count_alive_awaymission++
+	else
+		ts_count_alive_station++
+	// after 30 seconds, assuming nobody took control of it yet, offer it to ghosts.
+	spawn(150) // deciseconds!
```

Comment:
You don't need to comment that it's deciseconds; this should be understood. Also, use the `addtimer` system, here.